### ISSUE TITLE
entry point not found when using cimgui.dylib from docking branch

### ISSUE THREAD

**Issue by wihu** [cimgui/cimgui#93]
Hi, Im trying to build cimgui with dear imgui docking branch, followed the compile instructions with cmake to generate unix makefile and run make in osx terminal.
It spits out cimgui.dylib, however, when I tried to call some docking related functions e.g. igIsWindowDocked or igSetNextWindowDockID, i get entry point not found exception (im using ImGui.NET wrapper for my project).

I tried to dump dylib entry points:
[cimgui_entries.txt](https://github.com/cimgui/cimgui/files/3038023/cimgui_entries.txt)

and it looks like some functions has prefix like _ZXX, I think I missed out something when generating the makefile, im still wondering what the issue is, please kindly advise.

and perhaps the build instructions can be improved, so as to make it clearer on how to generate correct output (particularly when building from other branch than master)?

**Comment by sonoro1234** [comment]
Could you find this function in the generated cimgui.h and cimgui.cpp?

**Comment by wihu** [comment]
yes i could, here are the files:
[cimgui.h.txt](https://github.com/cimgui/cimgui/files/3038196/cimgui.h.txt)
[cimgui.cpp.txt](https://github.com/cimgui/cimgui/files/3038198/cimgui.cpp.txt)
(I need to append with .txt otherwise github refused to upload them).

**Comment by sonoro1234** [comment]
__ZN5ImGui14IsWindowDockedEv is a cpp imgui function (decorated name) but no igIsWidowDocked

It seems that somehow you are compiling imgui docking branch but just standard cimgui.
Could you check that you are compiling the new generated cimgui and not the older one?
Is the new generated file really copied to the root cimgui folder?

**Comment by wihu** [comment]
ok, I just rebuilt the dylib again after running generator, you are right, now igIsWindowDocked is exported correctly and i can call the function from my project.
So the issue is solved, however, I got a bunch of warnings like this:
 warning: 'igGetWindowPos' has C-linkage specified, but returns
      user-defined type 'ImVec2' which is incompatible with C [-Wreturn-type-c-linkage]
CIMGUI_API ImVec2 igGetWindowPos(void);

it looks fine though when looking at the generated cs codes, the return value is converted to System.Numerics.Vector2, so might not be a big issue. thanks for your prompt answers!
